### PR TITLE
#159053947 Initial tests for existing functionalities

### PR DESCRIPTION
**Pull request by petermwash**
#### What does this PR do?
Setup and write initial tests for testing the existing functionalities.

#### Description of tasks to be completed
Write failing tests for the existing functionalities

#### How should this be manually tested?
```sh
$ git clone https://github.com/andela/ah-titans.git
```
```sh
$ git checkout ch-tests-existing-functionalities-159053947
```
```sh
$ cd ah-titans/authors
```

> Add **.env** file and copy **example.env** content.
> Add your local database configuration settings to your **.env** file

```sh
$ source .env
```
```sh
$ python3.7 -m venv venv
```
```sh
$ source venv/bin/activate
```
```sh
$ pip install -r requirements.txt
```
```sh
$ python manage.py migrate
```
```sh
$ python manage.py test
```

#### Any background context you want to provide?
No

#### What are the relevant Pivotal Tracker Stories?
[#159053947](https://www.pivotaltracker.com/story/show/159053947)

### CODE REVIEW

File: tests/test_auth.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            "username":"nerd",
+            "email":"nerd@nerd.com",
+            "password":"secret"
+            }
+
+        # Initialize client
+        self.client = APIClient()
+
+    def test_api_can_create_a_user(self):
+        """Test the api can succefully create a user."""
```
Comment (evansmusomi) [inline tests/test_auth.py]: This line reads "succefully" instead of "successfully". Please update.

File: tests/test_auth.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        self.client.post(
+            '/api/users/',
+            self.user_data
+        )
+        response = self.client.post(
+            '/api/users/login',
+            login_data,
+            format='json'
+        )
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
```
Comment (ridgekimani) [inline tests/test_auth.py]: 401 unauthorized means "you aren’t authenticated–either not authenticated at all or authenticated incorrectly–but please reauthenticate and try again". Therefore in case of wrong credentials a 400 should suffice. Same as the test with wrong email

File: tests/test_models.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+class ModelTestCase(TestCase):
+    """This class defines the test suite for the article model."""
+
+    def setUp(self):
+        """Define the test client and other test variables."""
+        self.uzer = User.objects.create(username="nerd", email="nerd@nerd.com")
+
+    def test_model_can_create_a_user(self):
+        """Test the user model can create a user."""
+        self.assertTrue(isinstance(self.uzer, User))
```
Comment (ridgekimani) [inline tests/test_models.py]: Uzer is not an english word. Use self.user or self._user

File: tests/test_models.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    def test_model_returns_readable_representation(self):
+        """Test a readable string is returned for the model instance."""
+        self.assertEqual(str(self.uzer), "nerd@nerd.com")
+
```
Comment (ridgekimani) [inline tests/test_models.py]: Add an end of line

Comment (petermwash) [reply tests/test_models.py]: Okay working on that

File: tests/test_models.py (hunk 3 above)
Comment (petermwash) [reply tests/test_models.py]: I will change that right away

File: tests/test_auth.py (hunk 2 above)
Comment (petermwash) [reply tests/test_auth.py]: Thanks for pointing that out, I will work on it

File: tests/test_auth.py (hunk 1 above)
Comment (petermwash) [reply tests/test_auth.py]: Thanks for pointing that out. I will correct it right away.

**Comment by petermwash** [comment]
I have made the corrections as indicated on the earlier comments.

**Comment by janetnim** [comment]
looks like you implemented the feedback. Good.

**Comment by ronaldndirangu** [comment]
I think you have tested to existing endpoints to a good degree. Keep up the good work. However, think you can improve the coverage by also testing some edge cases like when a user tries to register using an empty field.

**Comment by johnmusiu** [comment]
Looks good to me, all feedback implemented

**Comment by janetnim** [comment]
Also maybe test validation on the user details like what should happen when they input a password of less than 8 characters etc

**Comment by petermwash** [comment]
@janetnim Thanks for reminding me that. Now it is done.
